packages/central-server/__tests__/sync/CentralSyncManager.test.js

```diff
@@ -433,6 +437,384 @@ describe('CentralSyncManager', () => {
+            recordId: patientProgramRegistration.id,
+            recordData: expect.objectContaining({
+              date: '2025-04-22 00:00:00',
+            }),
+            recordUpdate: true,
+          }),
+        ]),
+      );
+    });
+
+    it('changelog handles sync tick boundary conditions correctly', async () => {
+      const BOUNDARY_SYNC_TICK = 20;
+      const AFTER_BOUNDARY_SYNC_TICK = 21;
+      await models.Setting.set('audit.changes.enabled', true);
+
+      // Set up initial data
+      await models.LocalSystemFact.set(FACT_CURRENT_SYNC_TICK, BOUNDARY_SYNC_TICK);
+      const facility = await models.Facility.create(fake(models.Facility));
+      const patient = await models.Patient.create(fake(models.Patient));
+      const program = await models.Program.create(fake(models.Program));
+      const clinician = await models.User.create(fakeUser());
+
+      const programRegistry = await models.ProgramRegistry.create({
+        ...fake(models.ProgramRegistry),
+        programId: program.id,
+      });
+      const patientProgramRegistration = await models.PatientProgramRegistration.create({
+        ...fake(models.PatientProgramRegistration),
+        programRegistryId: programRegistry.id,
+        clinicianId: clinician.id,
+        patientId: patient.id,
+        facilityId: facility.id,
+      });
+      await models.PatientFacility.create({
+        id: models.PatientFacility.generateId(),
+        patientId: patient.id,
+        facilityId: facility.id,
+      });
+
+      // Make a change exactly at the boundary sync tick
+      patientProgramRegistration.date = '2025-04-22 00:00:00';
+      await patientProgramRegistration.save();
+
+      const centralSyncManager = initializeCentralSyncManager({
+        sync: {
+          lookupTable: {
+            enabled: true,
+          },
+          maxRecordsPerSnapshotChunk: DEFAULT_MAX_RECORDS_PER_SNAPSHOT_CHUNKS,
+        },
+      });
+
+      // Update lookup table at the boundary tick
+      await centralSyncManager.updateLookupTable();
+
+      // Make a change immediately after the boundary
+      await models.LocalSystemFact.set(FACT_CURRENT_SYNC_TICK, AFTER_BOUNDARY_SYNC_TICK);
+      patientProgramRegistration.date = '2025-04-23 00:00:00';
+      await patientProgramRegistration.save();
+
+      // Start sync session
+      const { sessionId } = await centralSyncManager.startSession();
+      await waitForSession(centralSyncManager, sessionId);
+
+      await centralSyncManager.setupSnapshotForPull(
+        sessionId,
+        {
+          since: BOUNDARY_SYNC_TICK,
+          facilityIds: [facility.id],
+        },
+        () => true,
+      );
+
+      const outgoingChanges = await centralSyncManager.getOutgoingChanges(sessionId, {});
+      const patientProgramRegistrationChange = outgoingChanges.find(
+        (c) => c.recordType === 'patient_program_registrations',
+      );
+
+      // Verify that the change at the boundary tick is included
```

Comment:
Oh man it's so hard to keep track of when it should be inclusive vs. exclusive, but I would have expected this to be exclusive!